presto-main/src/test/java/com/facebook/presto/sql/planner/TestCanonicalPlanGenerator.java

```diff
@@ -171,6 +174,55 @@ public void testTableScan()
         assertDifferentCanonicalLeafSubPlan("SELECT totalprice FROM orders", "SELECT orderkey, totalprice FROM orders");
         assertDifferentCanonicalLeafSubPlan("SELECT * FROM orders", "SELECT orderkey, totalprice FROM orders");
     }
+    @Test
+    public void testTableScanAndProjectWithStrategy()
+            throws Exception
+    {
+        assertSameCanonicalLeafPlan("SELECT 1 from orders", "SELECT 1 from orders", CONNECTOR);
+        assertDifferentCanonicalLeafPlan("SELECT 1 from orders", "SELECT 2 from orders", CONNECTOR);
+        assertSameCanonicalLeafPlan("SELECT 1 from orders", "SELECT 2 from orders", REMOVE_CONSTANTS);
+
+        assertSameCanonicalLeafPlan(
+                "SELECT totalprice, custkey + (totalprice / 10) from orders",
+                "SELECT custkey + (totalprice / 10), totalprice from orders",
+                CONNECTOR);
+        assertDifferentCanonicalLeafPlan(
+                "SELECT totalprice, custkey + (totalprice / 10) from orders",
+                "SELECT custkey + (totalprice / 5), totalprice from orders",
+                CONNECTOR);
+        assertSameCanonicalLeafPlan(
+                "SELECT totalprice, custkey + (totalprice / 10) from orders",
+                "SELECT custkey + (totalprice / 5), totalprice from orders",
+                REMOVE_CONSTANTS);
+    }
+
+    @Test
+    public void testFilterWithStrategy()
```

Comment:
Can we create more tests which clearly differentiate and explain these canonicalization strategies?

Reply:
Can you be more specific, which part do you think is missing? 

This tests for scan, filter, project, agg cases. Also, this isn't a great place to test canonicalization because this tests canonicalization of final physical plan, meanwhile HBO will try to canonicalize at some step of logical planning. More tests for different plan nodes are added in next PR which focus on more plan nodes